Add versioning docs for tokens (#4922)
grommet/hpe-design-system: aries-site/src/pages/design-tokens/versioning.mdx

import { Box, Grid, Image } from 'grommet';
import { Example } from '../../layouts';

<Example
  caption="Dashboard with ^0.9.0 (existing theme) applied on left and ^1.0.0 (updated theme) applied on right"
  width="100%"
>
  <Grid columns={['flex', 'flex']} gap="medium" align="start">
    <Box>

>       <Image src="/hpe-design-tokens-v0.png" fit="contain" />

Author: Done.

    </Box>
    <Box>
      <Image src="/hpe-design-tokens-v1.png" fit="contain" />
    </Box>
  </Grid>
</Example>

## Production-ready versions

### ^1.0.0

This release provides production-ready support with styles in alignment with the updated HPE theme.

In the v2 Figma Component Library, this is the default theme.

### ^0.9.0

This release provides production-ready support with styles in alignment with the old HPE theme (grommet-theme-hpe v5).

> To ease adoption of design tokens, v0.9.0 allows teams with existing applications to adopt design tokens while maintaining styling that aligns with existing theming.

Reviewer: Wondering if we can pull this messaging into an "info" notification following the "In the v2 Figma Component Library, this can be opted into by changing variables to use the "v0" modes."

```suggestion
```

Author: Done locally.

Reviewer: Still seeing this come through in addition to the notification

In the v2 Figma Component Library, this can be opted into by changing variables to use the "v0" modes.

> 

Reviewer: ```suggestion

<Notification
  title="Recommendation"
  message="To ease adoption of hpe-design-tokens, it is recommended to use v0.9.0 for initial mapping of tokens to local components. This will achieve style parity with an application's current look and allow quality assurance against familiar styling. When ready, teams can simply upgrade to v1 to view the latest theme styles."
/>

```

Author: Done locally and committed.

> ### Breaking changes from v0.9.0 to v1.0.0

Reviewer: ```suggestion
### Migrating from v0.9.0 to v1.0.0
```

> There are no breaking API changes from v0.9.0 to v1.0.0. The only changes are value changes to reflect the updated HPE theme.

Reviewer: ```suggestion
v1 reflects styling of the updated HPE theme and only contains token value changes. There are no breaking API changes from v0.9.0 to v1.0.0.
```

## Semantic versioning

> HPE Design tokens follows the standards defined by semantic versioning with slight adaptations at the major version level to more appropriately align to hpe-design-tokens as a style delivery mechanism.

Reviewer: ```suggestion
HPE Design tokens follows the standards defined by semantic versioning with slight adaptations at the major version level to align with hpe-design-tokens as a style delivery mechanism.
```

> Given a version number MAJOR.MINOR.PATCH, we’ll increment:

Reviewer: Trying to avoid "we" and such refereneces

```suggestion
Given a version number MAJOR.MINOR.PATCH, increments will occur when:
```

**1. MAJOR version** when there is a significant change to HPE’s brand expression and/or there are incompatible changes to the API.

> Significant changes to the look and feel of HPE’s brand (even if no breaking changes to the namespace are made) will be signaled via a major change (the look and feel of the new major version is not compatible with the look and feel of the old).

Reviewer: ```suggestion
Significant changes to the look and feel of HPE’s brand &mdash; even if no breaking changes to the namespace are made &mdash; will be signaled via a major change (i.e. the look and feel of the new major version is not compatible with the look and feel of the previous).
```

> This will help to signal that if service A is on v1.y.z and service B is on v2.y.z, there will be a mismatch in overall experience.

Reviewer: ```suggestion
This signals that if service A is on v1.y.z and service B is on v2.y.z, there will be a mismatch in overall experience.
```

**2. MINOR version** when there is functionality added in a backward compatible manner. Examples include:

- Introducing a new token.
- Updating a token value in an expected, minor way. For example, slightly darkening or lightening a text color based on user testing/feedback.

> **3. PATCH version** when we make backward compatible bug fixes. Examples include:

Reviewer: ```suggestion
**3. PATCH version** when backward compatible bug fixes are made. Examples include:
```
